axum/src/routing/mod.rs

```diff
@@ -58,6 +58,12 @@ macro_rules! panic_on_err {
 pub(crate) struct RouteId(u32);
 
 /// The router type for composing handlers and services.
+///
+/// `Router<S>` means a router that is _missing_ a state of type `S` to be able
+/// to handle requests. Thus, only `Router<()>` (i.e. without missing state) can
+/// be passed to [`serve()`]. See [`Router::with_state()`] for more details.
+///
+/// [`serve()`]: crate::serve()
```

Comment:
Nit, we don't usually include the parentheses. Honestly, I didn't even know it's possible.

```suggestion
/// be passed to [`serve`]. See [`Router::with_state`] for more details.
///
/// [`serve`]: crate::serve
```